src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/TypeNameCheck.java

```diff
@@ -63,29 +63,54 @@
  * An example of how to configure the check is:
  * </p>
  * <pre>
- * &lt;module name="TypeName"/&gt;
+ * &lt;module name=&quot;TypeName&quot;/&gt;
+ * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * public interface FirstName {} // OK
+ * protected class SecondName {} // OK
+ * enum Third_Name {} // violation, name 'Third_Name' must match pattern '^[A-Z][a-zA-Z0-9]*$'
+ * private class FourthName_ {} // violation, name 'FourthName_'
+ *                              // must match pattern '^[A-Z][a-zA-Z0-9]*$'
  * </pre>
  * <p>
  * An example of how to configure the check for names that begin with
- * a lower case letter, followed by letters, digits, and underscores is:
+ * a lower case letter, followed by letters, digits, and underscores.
+ * Also, suppress the check from being applied to protected and private type:
  * </p>
  * <pre>
- * &lt;module name="TypeName"&gt;
- *   &lt;property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/&gt;
+ * &lt;module name=&quot;TypeName&quot;&gt;
+ *   &lt;property name=&quot;format&quot; value=&quot;^[a-z](_?[a-zA-Z0-9]+)*$&quot;/&gt;
+ *   &lt;property name=&quot;applyToProtected&quot; value=&quot;false&quot;/&gt;
+ *   &lt;property name=&quot;applyToPrivate&quot; value=&quot;false&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * public interface firstName {} // OK
+ * public class SecondName {} // violation, name 'Second_Name'
+ *                            // must match pattern '^[a-z](_?[a-zA-Z0-9]+)*$'
+ * protected class ThirdName {} // OK
+ * private class FourthName {} // OK
+ * </pre>
  * <p>
  * The following configuration element ensures that interface names begin with {@code "I_"},
  * followed by letters and digits:
  * </p>
  * <pre>
- * &lt;module name="TypeName"&gt;
- *   &lt;property name="format"
- *     value="^I_[a-zA-Z0-9]*$"/&gt;
- *   &lt;property name="tokens"
- *     value="INTERFACE_DEF"/&gt;
+ * &lt;module name=&quot;TypeName&quot;&gt;
+ *   &lt;property name=&quot;format&quot;
+ *     value=&quot;^I_[a-zA-Z0-9]*$&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot;
+ *     value=&quot;INTERFACE_DEF&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * public interface I_firstName {} // OK
+ * interface SecondName {} // violation, name 'SecondName'
+ *                         // must match pattern '^[a-z](_?[a-zA-Z0-9]+)*$'
```

Comment:
Regex in comment is different from one in config example

Reply:
Done